### ISSUE TITLE
Support for adaptive triggers and tactile feedback when using a Dualsence gamepad

### ISSUE THREAD

**Issue by Remisonl** [Ryochan7/DS4Windows#2034]
To begin with, I will say that I am incredibly grateful to you for your work. This is beyond words.

I'll get down to business. Unfortunately, my English is pretty bad and so far, I have not been able to figure out if there is support for adaptive triggers and tactile feedback when using a Dualsence gamepad, and if so, how to enable them?

In addition, it is not entirely clear in which games support for these functions is implemented and whether it is in principle. Please, if it's not difficult for you, tell me specific games where these functions are implemented as fully as possible. Or maybe the best compatibility is realized in synergy with sites like Steam, or Epic Games?

Please explain to the teapot :) Request not only to the developer, but also to the users.

**Comment by xv0t** [comment]
There is (and probably won't be) any official support for adaptive triggers on PC, unless developers explicitly implement it and Sony releases an official driver for PC that does (there is a Linux driver but it lacks adaptive trigger/haptic feedback support).

DS4Windows can use 3 effects for the triggers (Full Click, Rigid and Pulse) but these are not dynamic, meaning they won't react to in-game actions (i.e. different trigger effects when firing a shotgun or a submachinegun).

I assume that by tactile feedback you mean haptic feedback? Just as with adaptive triggers, there is no official support for it and will likely never be. You can simulate the effect by sending sound output to the rear speakers in the DualSense, but it only works with the controller plugged to the PC and it's nothing like what is experienced on the PS5. You can easily find videos about it on Youtube.

So in short, there is no way to have adaptive triggers and haptic feedback on PC yet.

**Comment by Remisonl** [comment]
I understood you. Thank you for help :)

**Comment by Ryusennin** [comment]
A few months ago, Microsoft expressed interest in haptic feedback for future Xbox controllers. Maybe some day haptic could be officially implemented in XInput and thus be accessible to DS4W.

**Comment by Reinbowsaur** [comment]
Simulated haptic/immersion feedback could be done by creating a field on the screen that looks for changes, say place a custom sized box on a pixel or area, and if it changes trigger x rumble, or if x color is detected at this pixel/area, change light bar to x.

**Comment by Masamune3210** [comment]
> A few months ago, Microsoft expressed interest in haptic feedback for future Xbox controllers. Maybe some day haptic could be officially implemented in XInput and thus be accessible to DS4W.

If I'm not mistaken, to that effect, Microsoft actually has support for rumble triggers at least, the problem is I think its only through the API offered by Store apps. Could be wrong though

**Comment by Remisonl** [comment]
Guys, can you answer 2 questions, please?

1. How to enable the implemented 3 trigger effects in DS4Windows? (Full Click, Rigid and Pulse);

2. How to simulate the effect by sending sound output to the rear speakers in the DualSense using DS4Windows?

**Comment by Ryochan7** [comment]
Without direct support being provided by games, there is pretty much no point trying to hack together something that won't feel right anyway.
